From 72166b0749e777701ea2fd6791fda22642892bad Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Tue, 6 Apr 2021 12:15:34 +0300
Subject: [PATCH 38/42] pci:s32gen1: Enable PCIe coherent transactions

Enable Outer coherency for all PCIe transactions on the AXI bus.
There are no Ncore reconfigurations needed, but to take advantage of it,
the Linux device-tree PCIe node should declare a "dma-coherent" property.

Upstream-Status: Pending

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Issue: ALB-6758
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/pcie_s32gen1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/pcie_s32gen1.c b/drivers/pci/pcie_s32gen1.c
index 237c9871fa..5e8cbe3895 100644
--- a/drivers/pci/pcie_s32gen1.c
+++ b/drivers/pci/pcie_s32gen1.c
@@ -622,9 +622,9 @@ bool s32_pcie_init(void __iomem *dbi, int id, bool rc_mode,
 	if (!s32_pcie_set_link_width(dbi, id, linkwidth))
 		return false;
 
-	/* Set domain to 0 and cache to 3 */
+	/* Configure transactions as Cacheable, Outer Shareable. */
 	s32_pcie_change_mstr_ace_cache(dbi, 3, 3);
-	s32_pcie_change_mstr_ace_domain(dbi, 0, 0);
+	s32_pcie_change_mstr_ace_domain(dbi, 2, 2);
 
 	/* Test value for coherency control reg */
 	debug("COHERENCY_CONTROL_3_OFF: 0x%08x\n",
-- 
2.25.1

